src/directives/offense/poisonRoom.ts

```diff
@@ -1,169 +1,188 @@
+				DirectiveControllerAttack.createIfNotPresent(this.room.controller.pos,'room');
+			}
+
+			// send fighters if needed
+			if(this.room && this.room.playerHostiles.length > 0) {
+				DirectiveOutpostDefense.createIfNotPresent(this.room.controller.pos,'room');
+			}
+		}
+	}
+	private suspendColony(roomName: string): string {
+		if (Overmind.colonies[roomName]) {
+			const colonyMemory = Memory.colonies[roomName] as ColonyMemory | undefined;
+			if (colonyMemory) {
+				colonyMemory.suspend = true;
+				Overmind.shouldBuild = true;
+				return `Colony ${roomName} suspended.`;
+			} else {
+				return `No colony memory for ${roomName}!`;
+			}
+		} else {
+			return `Colony ${roomName} is not a valid colony!`;
+		}
+	}
+
+	private poisonActions() {
+		if(!(this.room && this.room.controller && this.room.controller.level == 2 && 
+			 this.walkableControllerPosisions && this.walkableControllerPosisions)) {
+			return;
+		}
+
+		// Poison Controller First
+		if(this.walkableControllerPosisions.length > 0) {
+			_.first(this.walkableControllerPosisions).createConstructionSite(STRUCTURE_WALL);
+			return;
+		} 
+
+		// Poison Sources Next
+		// Trick: Csite are not walkable, wait for poisoner to carry energy before blocking Sources
+		// Trick: Remove sources Csites if poisoner need to harvest, i.e carrying 0 energy
+		if (this.overlords.roomPoisoner.roomPoisoners.length &&
+			this.overlords.roomPoisoner.roomPoisoners[0].carry.energy > 0) {
+				if(this.walkableSourcePosisions.length) {
+					_.first(this.walkableSourcePosisions).createConstructionSite(STRUCTURE_WALL);
+				}
+		} else {
+				_.forEach(this.room.constructionSites, csite => {csite.remove();} );
+		}
+		
+	}
+	private isPoisoned(): boolean {
+		let result = false;
+		if (this.room && this.room.controller!.level > 1) {
+			result = !!this.walkableSourcePosisions && !this.walkableSourcePosisions.length &&
+					 !!this.walkableControllerPosisions && !this.walkableControllerPosisions.length;
+			return result;
+		} else {
+			return false;
+		}
+	}
+
+	private prePoisonActions(): void {
+		if(!(this.room && this.room.controller && this.room.controller.my)) {
+			return;
+		}
+		// kill claimer if room claimed, it can blocking csite creation
+		// it can also be used as a quick energy source from the tomb
+		if (this.overlords.claim.claimers.length) {
+			this.overlords.claim.claimers[0].suicide();
+		}
+		// remove any containers that can be next to sources
+		if(this.room.containers.length) {
+			_.forEach(this.room.containers, container => {container.destroy();});
+		}
+		// remove any hostile consituction sites
+		_.forEach(this.room.find(FIND_HOSTILE_CONSTRUCTION_SITES), csite => {csite.remove();});
+	}
+
+	private unclaimActions(): void {
+		if(!(this.room && this.room.controller)) {
+			return;
+		}
+		// remove roads before unclaiming
+		if(this.room.roads.length > 0) {
+			_.forEach(this.room.roads, road => {road.destroy();} );
+		}
```

Comment:
Nice 👿

Reply:
can you build walls on top of roads? if not, then if there are roads next to something you want to block off then you won't be able to build walls on top of them. you should remove all non-wall structures in the room during your `poisonActions` call to avoid this.

Reply:
yes, I can build walls on top of roads, have done it few days ago.
but u r right, may be I should just remove all structures except for walls but keep storage/terminals with resources in them (as a default behavior)